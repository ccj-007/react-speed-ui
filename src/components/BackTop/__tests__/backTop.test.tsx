import React from 'react'
import { render, screen } from '@testing-library/react'
import '@testing-library/jest-dom'
import { BackTop } from '../../index'

describe('test backTop component', () => {
  it('should render component', () => {
    const Demo = () => {
      const boxRef = React.useRef<HTMLDivElement>(null)
      return (
        <>
          <div ref={boxRef}></div>
          <BackTop container={boxRef} visibilityHeight={100}>容器内 backTop</BackTop>
          <BackTop visibilityHeight={300}>root层级 backTop</BackTop>
        </>
      )
    }
    const { asFragment } = render(
      <Demo></Demo>
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
