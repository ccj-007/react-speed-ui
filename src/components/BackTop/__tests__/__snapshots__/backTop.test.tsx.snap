// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test backTop component should render component 1`] = `
<DocumentFragment>
  <div />
</DocumentFragment>
`;
